gateway/src/apicast/backend_client.lua

```diff
@@ -142,6 +145,32 @@ local function authorize_options(using_oauth)
   return { headers = headers }
 end
 
+local function add_transaction(transactions, index, cred_type, cred, reports)
+  local index_with_cred = format('transactions[%s][%s]', index, cred_type)
+  transactions[index_with_cred] = cred
+
+  for metric, value in pairs(reports) do
+    local index_with_metric = format('transactions[%s][usage][%s]', index, metric)
+    transactions[index_with_metric] = value
+  end
+end
+
+local function format_transactions(reports_batch)
```

Comment:
I'm not sure this method should live in the backend client.
It depends on ReportsBatch.

IMO this client should be unaware of any of other objects and just work with plain tables (and possibly __tostring metamethod on them).